### PR TITLE
NIFI-9483 Exclude Log4j 2 log4j-core and related libraries

### PR DESCRIPTION
**Pull request by exceptionfactory**
#### Description of PR

NIFI-9483 Excludes Log4j 2 `log4j-core` and related Log4j 2 libraries from the following extension modules that do not have runtime dependencies on Log4j 2:

- nifi-atlas-reporting-task
- nifi-druid-controller-service-api
- nifi-hive3-processors
- nifi-ranger-plugin
- nifi-registry-ranger-plugin

The `nifi-elasticsearch-5-processors` module requires Log4j 2 `log4j-api` to support Elasticsearch at runtime. Instead of `log4j-core`, the `log4j-to-slf4j` library provides runtime routing of Log4j 2 Loggers to SLF4J, removing the need for `log4j-core`.

These changes remove the need for any references to the Log4j 2 `log4j-core` library.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [X] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -795,8 +795,8 @@
                                         <exclude>com.google.code.findbugs:jsr305:*:*:compile</exclude>
                                         <!-- Log4J excluded in favor of log4j-over-slf4j and logback -->
                                         <exclude>log4j:log4j:*</exclude>
-                                        <!-- Ban log4j-core less than 2.15.0 due to Log4Shell vulnerability -->
-                                        <exclude>org.apache.logging.log4j:log4j-core:(,2.15.0)</exclude>
+                                        <!-- Log4j 2 log4j-core excluded in favor of log4j-to-slf4j routing to logback -->
```
Comment (ChrisSamo632) [inline pom.xml:798]: While it wasn't directly past of the jira, it's it worth updating log4j to 2.17.0 to address CVE-2021-45105 (a DoS vulnerability found after the release of 2.16.0)?

**Review by ChrisSamo632 — APPROVED** [review]

**Comment by ChrisSamo632** [comment]
@exceptionfactory needs rebasing after https://github.com/apache/nifi/pull/5615 merged - otherwise 👍 and happy to merge once the conflicts are resolved.

**Comment by exceptionfactory** [comment]
Thanks for the feedback and approval @ChrisSamo632! I rebased to resolve the conflict, as expected, with the banning of `log4j-core`.

**Comment by joewitt** [comment]
+1 merged to main.  Thanks!

**Pull request closed without merging**